osu.Game.Rulesets.Taiko/UI/DrumTouchInputArea.cs

```diff
@@ -142,15 +151,56 @@ private void handleUp(object source)
         private bool validMouse(MouseButtonEvent e) =>
             leftRim.Contains(e.ScreenSpaceMouseDownPosition) || rightRim.Contains(e.ScreenSpaceMouseDownPosition);
 
-        private TaikoAction getTaikoActionFromInput(Vector2 inputPosition)
+        private TaikoAction getTaikoActionFromInput(TaikoAction input)
+        {
+            switch (configTouchControlScheme.Value)
+            {
+                case TaikoTouchControlScheme.KDDK:
+
+                    switch (input)
+                    {
+                        case TaikoAction.LeftRim: return TaikoAction.LeftRim;
+                        case TaikoAction.LeftCentre: return TaikoAction.LeftCentre;
+                        case TaikoAction.RightCentre: return TaikoAction.RightCentre;
+                        case TaikoAction.RightRim: return TaikoAction.RightRim;
+                    }
+                    break;
+
+                case TaikoTouchControlScheme.DDKK:
+                    switch (input)
+                    {
+                        case TaikoAction.LeftRim: return TaikoAction.LeftCentre;
+                        case TaikoAction.LeftCentre: return TaikoAction.RightCentre;
+                        case TaikoAction.RightCentre: return TaikoAction.LeftRim;
+                        case TaikoAction.RightRim: return TaikoAction.RightRim;
+                    }
+                    break;
+
+                case TaikoTouchControlScheme.KKDD:
+                    switch (input)
+                    {
+                        case TaikoAction.LeftRim: return TaikoAction.LeftRim;
+                        case TaikoAction.LeftCentre: return TaikoAction.RightRim;
+                        case TaikoAction.RightCentre: return TaikoAction.LeftCentre;
+                        case TaikoAction.RightRim: return TaikoAction.RightCentre;
+                    }
+                    break;
+            }
```

Comment:
The reason that this remapping is so confusing is that it's misguided in the first place. Don't remap a type onto itself, nobody will understand this. Use a numerical index instead or something, as in

| style | button 0 | button 1 | button 2 | button 3 |
| :-: | :-: | :-: | :-: | :-: |
| kddk | left rim | left centre | right rim | right centre |
| kkdd | left centre | right centre | left rim | right rim |
| ddkk | left rim | right rim | left centre | right centre |